contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -742,51 +768,176 @@ have a built-in `nil` value.
+    values during the API call
+ 3. controllers: arbitrary changes (within the bounds of what is allowed) can
+    be made to a resource after the API call has completed
+
+Some care is required when deciding which mechanism to use and managing the
+semantics.
+
+### Static Defaults
+
+Static default values are specific to each API version.  The default field
+values applied when creating an object with the "v1" API may be different than
+the values applied when using the "v2" API.  In most cases, these values are
+defined as literal values by the API version (e.g. "if this field is not
+specified it defaults to 0").
+
+In some cases, these values may be conditional on or deterministically derived
+from other fields (e.g. "if otherField is X then this field defaults to 0" or
+"this field defaults to the value of otherField").  Note that such derived
+defaults present a hazard in the face of updates - if the "other" field
+changes, the derived field may have to change, too.  The static defaulting
+logic is unaware of updates and has no concept of "previous value", which means
+this inter-field relationship becomes the user's problem - they must update
+both the field they care about and the "other" field.
+
+In very rare cases, these values may be allocated from some pool or determined
+by some other method (e.g. Service's IP and IP-family related fields need to
+consider other configuration settings).
+
+These values are applied synchronously by the API server when decoding
+versioned data.  For CREATE and UPDATE operations this is fairly
+straight-forward.  When the API call completes, all static defaults will have
+been set and stored.  Subsequent GETs of the resource will include the default
+values explicitly.  However, static defaults also apply when an object is read
+from storage (i.e. GET operations).  This means that when someone GETs an
+"older" stored object, any newly added fields will be defaulted and returned
+according to the API version that is stored.
+
+Static defaults are the best choice for values which are logically required,
+but which have a value that works well for most users.  Static defaulting
+must not consider any state except the object being operated upon (and the
+complexity of Service API stands as an example of why).
 
 Default values can be specified on a field using the `+default=` tag. Primitives
 will have their values directly assigned while structs will go through the
 JSON unmarshalling process. Fields that do not have an `omitempty` json tag will
 default to the zero value of their corresponding type if no default is assigned.
 
-Refer to [defaulting docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting) for more information.
+Refer to [defaulting docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting)
+for more information.
+
+### Admission Controlled Defaults
```

Comment:
Clarify ordering here?  When a object is created and a field is left unset, does admission control see that the field is unset or is the default already filled in?

Reply:
Will add text in next push